### ISSUE TITLE
No logging of Sabre\DAV\Exception\InsufficientStorage as fatal if quota is reached

### ISSUE THREAD

**Issue by alx-tuilmenau** [nextcloud/server#20514]
Feature Request:

Is there a possibility to change the logging of "Sabre\DAV\Exception\InsufficientStorage: Insufficient space in ..." from fatal to warning or info, if only the quota for a user is full, so the logging of these messages can be turned off without missing real errors ?

The message is logged, if a user tries to upload further files via Desktop Client, if his quota is full (and a lot of times, because the client is still trying every file...)

**Comment by szaimen** [comment]
Is this Issue still valid in NC21.0.2? If not, please close this issue. Thanks! :)

**Comment by alx-tuilmenau** [comment]
Sorry, I missed the comment. Its still present in 21.0.3. The Message is only logged for Uploads via Client / WebDAV, uploading via Web gives the user a message and no message in Log.

**Comment by szaimen** [comment]
cc @nextcloud/server-triage is this request feasible?

**Comment by ThomasCr** [comment]
This is still present in current NC 24.0.6.

**Comment by alx-tuilmenau** [comment]
Yes, it's still present. Still got tons of log entries if someone's client is trying to sync with full quota.

**Comment by moboter** [comment]
Interesting enough I have the issue in nextcloud 27.1.9 with the desktopclient version 3.13.0
Quota was reached one Than i extended the quota I can upload files via webbrowser but the client isnt syncing there are still 140 GB of space available for the root partition,
groupfolders:list tells me that Quota | Size  are 38/16 GB
Its on a local machine Ubuntu 22 PHP 8.1.2 -MariaDB 10.6.16

**Comment by alx-tuilmenau** [comment]
I also got still this Message with 28.0.6:
<details>
<summary>Error Message in log</summary>
<code>
{
  "reqId": "iqPstuYSWWfOWp6vWUmt",
  "level": 3,
  "time": "2024-05-29T10:02:25+00:00",
  "remoteAddr": "xxx",
  "user": "kxxx",
  "app": "webdav",
  "method": "MOVE",
  "url": "/remote.php/dav/uploads/xxx/2114599060/.file",
  "message": "Insufficient space in xxx, 326966101716 required, 105978238676 available",
  "userAgent": "Mozilla/5.0 (Windows) mirall/3.4.1stable-Win64 (build 20211221) (Nextcloud, windows-10.0.22631 ClientArchitecture: x86_64 OsArchitecture: x86_64)",
  "version": "28.0.6.1",
  "exception": {
    "Exception": "Sabre\\DAV\\Exception\\InsufficientStorage",
    "Message": "Insufficient space in /xxx required, 105978238676 available",
    "Code": 0,
    "Trace": [
      {
        "file": "/data/nextcloud_a3/apps/dav/lib/Connector/Sabre/QuotaPlugin.php",
        "line": 142,
        "function": "checkQuota",
        "class": "OCA\\DAV\\Connector\\Sabre\\QuotaPlugin",
        "type": "->"
      },
      {
        "file": "/data/nextcloud_a3/3rdparty/sabre/event/lib/WildcardEmitterTrait.php",
        "line": 89,
        "function": "beforeMove",
        "class": "OCA\\DAV\\Connector\\Sabre\\QuotaPlugin",
        "type": "->"
      },
      {
        "file": "/data/nextcloud_a3/3rdparty/sabre/dav/lib/DAV/CorePlugin.php",
        "line": 603,
        "function": "emit",
        "class": "Sabre\\DAV\\Server",
        "type": "->"
      },
      {
        "file": "/data/nextcloud_a3/3rdparty/sabre/event/lib/WildcardEmitterTrait.php",
        "line": 89,
        "function": "httpMove",
        "class": "Sabre\\DAV\\CorePlugin",
        "type": "->"
      },
      {
        "file": "/data/nextcloud_a3/3rdparty/sabre/dav/lib/DAV/Server.php",
        "line": 472,
        "function": "emit",
        "class": "Sabre\\DAV\\Server",
        "type": "->"
      },
      {
        "file": "/data/nextcloud_a3/3rdparty/sabre/dav/lib/DAV/Server.php",
        "line": 253,
        "function": "invokeMethod",
        "class": "Sabre\\DAV\\Server",
        "type": "->"
      },
      {
        "file": "/data/nextcloud_a3/3rdparty/sabre/dav/lib/DAV/Server.php",
        "line": 321,
        "function": "start",
        "class": "Sabre\\DAV\\Server",
        "type": "->"
      },
      {
        "file": "/data/nextcloud_a3/apps/dav/lib/Server.php",
        "line": 373,
        "function": "exec",
        "class": "Sabre\\DAV\\Server",
        "type": "->"
      },
      {
        "file": "/data/nextcloud_a3/apps/dav/appinfo/v2/remote.php",
        "line": 35,
        "function": "exec",
        "class": "OCA\\DAV\\Server",
        "type": "->"
      },
      {
        "file": "/data/nextcloud_a3/remote.php",
        "line": 172,
        "args": [
          "/data/nextcloud_a3/apps/dav/appinfo/v2/remote.php"
        ],
        "function": "require_once"
      }
    ],
    "File": "/data/nextcloud_a3/apps/dav/lib/Connector/Sabre/QuotaPlugin.php",
    "Line": 216,
    "message": "Insufficient space in xxx, 326966101716 required, 105978238676 available",
    "exception": [],
    "CustomMessage": "Insufficient space in /xxx, 326966101716 required, 105978238676 available"
  },
  "id": "66571b0184b4b"
}
</code>
</details>

**Comment by Justinzobel** [comment]
Can this please be looked into. Displaying errors to admins that aren't relevant can cause notification/alert fatigue and they can begin to ignore looking at the logs because it's "just noise".

**Comment by Justinzobel** [comment]
Optionally split this into two functions, one for user-quota not enough storage and another for server not enough storage.

As the second one is critical information to the server administrator. The second is irrelevant to the administrator and only relevant to the user.

**Comment by Semellie** [comment]
I would also request this get looked at. Log entries on the error level that are practically irrelevant to me as an admin should not happen.
